src/callbacks/checkpoint.cpp

```diff
@@ -289,15 +291,15 @@ std::string checkpoint::find_latest_checkpoint(model *m, std::string& latest_fil
   // Update other ranks on where we are loading from.
   // TODO: we would want to prepend dir with the model name and model rank:
   // m->get_name() + '.' + std::to_string(comm->get_trainer_rank()) + '.'
-#if 1
   header_t<max_len_dirname> header;
+  bzero(&header, sizeof(header_t<max_len_dirname>));
```

Comment:
Use `std::memset` instead. `bzero` is a legacy function, the use of which is deprecated.
```suggestion
  std::memset(&header, 0x0, sizeof(header_t<max_len_dirname>));
```